### DISCUSSION TITLE
Start copilot default disabled

### DISCUSSION THREAD

**Discussion by ciuckc** [LazyVim/LazyVim#1151] [Q&A] [answered]
Hi is there a way to start  with copilot disabled as default?

**Comment by skoch13** [comment]
Simply comment this extra line in your lazy.lua and Copilot will be disabled

```
...
 -- { import = "lazyvim.plugins.extras.coding.copilot" },
...

```

**Reply by ciuckc** [reply]
I meant it in the sense where you don t have to type ':Copilot disable' and just enable it when you need it?

**Comment by mirsella** [comment, marked as answer]
`vim.cmd("silent! Copilot disable")`

**Reply by ciuckc** [reply]
This works nicely!
Heloo from 42 amsterdam!

**Reply by weberam2** [reply]
Where did you put this?

**Reply by mirsella** [reply]
plugin/copilot.nvin and in function config maybe, or in autocmd.lua

**Reply by weberam2** [reply]
This is still not clear to me
Or is not working...

**Reply by meleu** [reply]
Trying to make it a bit clearer...

I've put this in my `lua/plugins/copilot.lua` (create the file if it doesn't exist):
```lua
return {
  -- start vim with copilot disabled by default
  vim.cmd("silent! Copilot disable"),
}
```

**Reply by parisni** [reply]
above did raise an error. So I did put this in `lua/config/autocmds.lua`
```lua
-- start vim with copilot disabled by default
vim.cmd("silent! Copilot disable")
```

**Comment by codeRSH** [comment]
Hello, 

I have used the following statement in autocmd.lua

vim.cmd("silent! Copilot disable")

it does disable the copilot by default, but I keep getting this irritating Warning message everytime I open Lazyvim. 

![image](https://github.com/user-attachments/assets/28122636-bf65-4501-aa8c-3179b5463479)

I suspect this is happening due to the default LazyVim Copilot.lua config. Is there a way I can disable or at least auto hide this particular warning message. Please help!

**Reply by dpetka2001** [reply]
That's a default warning that `copilot.lua` sends when the plugin is loaded and the client is not attached. 

You can add 
```lua
      logger = {
        print_log_level = vim.log.levels.ERROR, -- or `OFF` instead of `ERROR` if you want to disable logging as a whole
      },
```
to your Copilot personal configuration. Read [copilot docs](https://github.com/zbirenbaum/copilot.lua#setup-and-configuration) for more info.

**Reply by codeRSH** [reply]
Thanks so much, this helped! 

FYI for anyone else have very similar issue with Codeium: 
![image](https://github.com/user-attachments/assets/0dffe55f-d51d-4949-9e1e-99bd1e448040)

I solved it by referring the Codeium (now Windsurf) documentation and I set the following environment variable in my zshrc (or bashrc) file: 
export DEBUG_CODEIUM=warn

So now normal 'info' messages will not show.
